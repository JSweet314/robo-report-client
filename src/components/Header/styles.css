header {
  padding: .8em;
  display: flex;
  width: 95%;
  align-items: center;
  justify-content: space-between;
}

header h1 {
  font-size: 2.6em; 
}

header nav {
  font-size: 1.5em;
  width: 25vw;
  display: flex;
  justify-content: space-between;
}

.hidden {
  display: none;
}
