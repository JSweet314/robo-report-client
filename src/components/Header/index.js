import React from 'react';
import PropTypes from 'prop-types';
import { NavLink, Link } from 'react-router-dom';
import logo from '../../assets/images/no-cellphone.svg';
import './styles.css';

const Header = ({ user, handleOAuthSignIn }) => {
  const myAccountNavLink = (
    <NavLink to="/myAccount" className="header__link">
      MY ACCOUNT
    </NavLink>
  );

  const signInButton = (
    <button
      className="header__link header__link--sign-in-btn"
      onClick={handleOAuthSignIn}
    >
      SIGN IN
    </button>
  );

  const myReportsNavLink = (
    <NavLink
      to="/myReports"
      className="header__link header__link--reports-button"
    >
      MY REPORTS
    </NavLink>
  );

  const accountButton = user.id ? myAccountNavLink : signInButton;

  const myReportsButton = user.id ? myReportsNavLink : null;
  return (
    <header className="header">
      <Link to="/" className='header__logo-group'>
        <img src={logo} alt="no phone calls"/>
        <h1 className="header__header">Robo Report</h1>
      </Link>
      <nav className="header__nav">
        <NavLink exact to="/" className="header__link header__link--home">
          HOME
        </NavLink>
        {myReportsButton}
        {accountButton}
      </nav>
    </header>
  );
};

Header.propTypes = {
  user: PropTypes.object.isRequired,
  handleOAuthSignIn: PropTypes.func.isRequired
};

export default Header;
