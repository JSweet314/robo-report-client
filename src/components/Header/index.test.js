import React from 'react';
import { shallow } from 'enzyme';
import Header from './';

describe('Header', () => {
  let wrapper;
  const mockHandleOAuthSignIn = jest.fn();
  
  beforeEach(() => {
    wrapper = shallow(
      <Header 
        isLoggedIn={false}
        handleOAuthSignIn={mockHandleOAuthSignIn}
      />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot', () => {
    wrapper = shallow(
      <Header 
        isLoggedIn={true}
        handleOAuthSignIn={mockHandleOAuthSignIn}
      />
    );
    
    expect(wrapper).toMatchSnapshot();
  });
});