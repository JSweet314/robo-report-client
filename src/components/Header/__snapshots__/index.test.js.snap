// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot 1`] = `
<header>
  <Link
    replace={false}
    to="/"
  >
    <h1>
      Robo Report
    </h1>
  </Link>
  <nav>
    <Link
      replace={false}
      to="/"
    >
      Home
    </Link>
    <NavLink
      activeClassName="active"
      ariaCurrent="true"
      className="hidden"
      to="/myReports"
    >
      My Reports
    </NavLink>
    <button
      className="sign-in--button"
      onClick={[Function]}
    >
      Sign In
    </button>
  </nav>
</header>
`;

exports[`Header should match the snapshot 2`] = `
<header>
  <Link
    replace={false}
    to="/"
  >
    <h1>
      Robo Report
    </h1>
  </Link>
  <nav>
    <Link
      replace={false}
      to="/"
    >
      Home
    </Link>
    <NavLink
      activeClassName="active"
      ariaCurrent="true"
      className="display"
      to="/myReports"
    >
      My Reports
    </NavLink>
    <button
      className="sign-in--button"
      onClick={[Function]}
    >
      Sign Out
    </button>
  </nav>
</header>
`;
